c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -113,31 +110,13 @@ impl<'tcx> TypeChecker<'tcx, '_> {
                     panic!("Creating non-null pointers from exposed addresses not supported");
                 }
             }
-            CastKind::Pointer(PointerCast::Unsize) => {
-                let pointee_to_ty = to_ty
-                    .builtin_deref(true)
-                    .unwrap_or_else(|| panic!("unsize cast has non-pointer output {:?}?", to_ty))
-                    .ty;
-
-                assert_matches!(from_lty.kind(), TyKind::Ref(..) | TyKind::RawPtr(..));
-                let pointee_from_lty = assert_matches!(from_lty.args, [lty] => lty);
-
-                let elem_to_ty = assert_matches!(pointee_to_ty.kind(), &TyKind::Slice(ty) => ty);
-                assert!(matches!(pointee_from_lty.kind(), TyKind::Array(..)));
-                let elem_from_lty = assert_matches!(pointee_from_lty.args, [lty] => lty);
-                assert_eq!(elem_from_lty.ty, elem_to_ty);
-                assert_eq!(pointee_from_lty.label, PointerId::NONE);
-                self.do_assign_pointer_ids(to_lty.label, from_lty.label);
-            }
-            CastKind::Pointer(..) => {
-                // The source and target types are both pointers, and they have identical pointee types.
-                // TODO: remove or move check to `is_castable_to`
-                assert!(from_lty.args[0].ty == to_lty.args[0].ty);
-                assert!(is_castable_to(from_lty, to_lty));
-            }
             _ => {
-                // A cast such as `T as U`
-                assert!(is_castable_to(from_lty, to_lty));
+                // TODO add dataflow constraints
```

Comment:
As the dataflow constraints are already not done correctly/at all prior to this PR, I'm not going to fix that here yet, but will fix that in a follow-up PR so that this one doesn't get too large (it's already very old).

Reply:
Please open an issue for this and make it a high priority.  Debugging this sort of constraint-solving analysis is hard enough already - we shouldn't knowingly leave it in a broken state on top of that.

Reply:
I will.  I was planning on working on this next.

Reply:
Tracked in #913.